security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -846,7 +847,14 @@ func (sc *SecretCache) generateSecret(ctx context.Context, token string, connKey
 		return nil, err
 	}
 
+	workloadIdentity := &spiffe.Identity{
+		TrustDomain:    sc.configOptions.TrustDomain,
+		Namespace:      sc.configOptions.WorkloadNamespace,
+		ServiceAccount: sc.configOptions.ServiceAccount,
+	}
+
 	options := pkiutil.CertOptions{
+		Host:       workloadIdentity.String(),
```

Comment:
What happens if host here does not match the identity of the pod?

Reply:
My concern is a pod overriding an env var and assuming a different identity

Reply:
How about Istiod directly verify caller identities (extract from JWT) and URI from CSR ?

Reply:
I do not see where this is done?

Please do not resolve comments if they are not addressed.